src/app/forms/custom-sections/adr-tank-statement-un-number-edit/adr-tank-statement-un-number-edit.component.html

```diff
@@ -1,40 +1,44 @@
-<ng-container *ngIf="control && formArray.controls[0]">
-  <app-field-error-message
-    [error]="!formArray.controls[0].value && control.invalid && (control.touched || submitted) ? control.meta.customErrorMessage : ''"
-  ></app-field-error-message>
+<ng-container *ngIf="control && formArray.controls.length">
+  <app-field-error-message [error]="control.touched ? control.getError('custom')?.message : ''"></app-field-error-message>
 
-  <ng-container *ngIf="formArray.controls.length === 1">
+  <ng-container *ngIf="formArray.controls.length === 1; else multiple">
     <app-text-input
-      [id]="control.meta.name"
-      [name]="formArray.controls[0].meta.name"
-      [label]="formArray.controls[0].meta.label + ' 1'"
-      [customId]="formArray.controls[0].meta.customId"
+      [id]="'UN_number_1'"
+      [name]="'UN number 1'"
+      [label]="'UN number 1'"
+      [customId]="'UN number 1'"
       [formControl]="formArray.controls[0]"
+      [minlength]="1"
+      [maxlength]="1500"
       (blur)="formArray.controls[0].markAsTouched(); control.markAsTouched()"
     ></app-text-input>
   </ng-container>
 
-  <!-- Additional UN numbers -->
-  <table class="govuk-table" *ngIf="formArray.controls.length > 1">
-    <tbody class="govuk-table__body">
-      <ng-container *ngFor="let child of formArray.controls; let i = index">
-        <tr>
-          <td class="govuk-table__cell">
-            <app-text-input
-              [name]="child.meta.name"
-              [label]="(child.meta.label ?? '') + ' ' + (i + 1)"
-              [customId]="child.meta.customId"
-              [formControl]="child"
-              (blur)="child.markAsTouched(); control.markAsTouched()"
-            ></app-text-input>
-          </td>
-          <td class="govuk-table__cell align-middle">
-            <a class="govuk-link" role="button" href="javascript:void(0)" (click)="removeControl(i)">Remove</a>
-          </td>
-        </tr>
-      </ng-container>
-    </tbody>
-  </table>
+  <ng-template #multiple>
+    <table class="govuk-table">
+      <tbody class="govuk-table__body">
+        <ng-container *ngFor="let child of formArray.controls; let i = index">
+          <tr>
+            <td class="govuk-table__cell">
+              <app-text-input
+                [id]="'UN_number_' + (i + 1)"
```

Comment:
Will this not duplicate the first ID? seen as you have UN_number_1 set on line 6?

Reply:
obviously the same goes for the other fields

Reply:
isn't it an if else? so the top one won't render

Reply:
This is part of an if-else statement (done in Angular's control flow using *ngIf="con; else ngTemplate" - essentially we show a standalone UN number 1 when there is 1 control in the form array because this one should have a remove button and thus is full width - and no it a table. If we have more than 1 we render them all in a table, and all controls, including UN number 1 has a remove link.

Reply:
Ah right sorry missed that